File: fluttermultiplatform/lib/api/auth_api.dart
import 'api_client.dart';

class AuthApi {
  static Future register(Map<String, dynamic> data) => api.post('/auth/register', data: data);
  static Future login(Map<String, dynamic> data) => api.post('/auth/login', data: data);
  static Future getProfile() => api.get('/auth/profile');
  static Future updateProfile(Map<String, dynamic> data) => api.put('/auth/profile', data: data);
}

File: frontend/src/api/authApi.ts
import api from './axios';
export const authApi = {
    register: (data: { name: string; email: string; password: string; phone?: string; role: string }) =>
        api.post('/auth/register', data),
    login: (data: { email: string; password: string }) =>
        api.post('/auth/login', data),
    getProfile: () =>
        api.get('/auth/profile'),
    updateProfile: (data: { name?: string; phone?: string; address?: string; avatar?: string }) =>
        api.put('/auth/profile', data),
};
